runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -182,33 +152,867 @@ fn test_pallet_xcm_recognize_multilocation() {
 			0
 		));
 		assert_eq!(Balances::free_balance(&AccountId::from(ALICE)), 499_999_000_000_000_000);
+		assert_eq!(Balances::free_balance(&sibling_account(2)), 0);
 		assert_eq!(
 			last_event(),
-			// XCMP_QUEUE SendError Transport
+			// Not XCMP_QUEUE in production environment
+			// This is the error of mimic XcmRouter: decl_test_network
 			Event::PolkadotXcm(pallet_xcm::Event::Attempted(Outcome::Incomplete(
-				200_000_000,
-				XcmError::Transport("")
+				UnitWeightCost::get(),
+				XcmError::Unroutable
 			)))
 		);
 		assert_ok!(PolkadotXcm::reserve_transfer_assets(
 			Origin::signed(AccountId::from(ALICE)),
 			Box::new((Parent, Parachain(2)).into()),
-			Box::new((Junction::AccountId32 { network: Any, id: ALICE }).into().into()),
+			Box::new((Junction::AccountId32 { network: Any, id: BOB }).into().into()),
 			Box::new(
 				vec![MultiAsset {
 					id: AssetId::Concrete(
 						CurrencyIdMultiLocationConvert::convert(CurrencyId::SelfReserve).unwrap(),
 					),
-					fun: Fungibility::Fungible(1_000_000_000_000),
+					fun: Fungibility::Fungible(2_000_000_000_000),
+				}]
+				.into()
+			),
+			0
+		));
+		assert_eq!(Balances::free_balance(&AccountId::from(ALICE)), 499_997_000_000_000_000);
+		assert_eq!(
+			Balances::free_balance(&sibling_account(2)),
+			2_000_000_000_000 // Only non trpped asset is in sovereign account
+		);
+		assert_eq!(
+			last_event(),
+			Event::PolkadotXcm(pallet_xcm::Event::Attempted(Outcome::Complete(
+				UnitWeightCost::get()
+			)))
+		);
+	});
+
+	ParaB::execute_with(|| {
+		assert_eq!(
+			Tokens::free_balance(
+				1, // Asset_id=1. The first registered Token in Para B
+				&AccountId::from(BOB)
+			),
+			u128::from(2_000_000_000_000 - UnitWeightCost::get() * 4)
+		);
+	});
+	// Notice so far pallet_xcm does not handle the asset transfer back - 0.9.23
+}
+
+#[test]
+fn test_methods_xtokens_expected_succeed() {
+	relaychain_parachains_set_up();
+	ParaA::execute_with(|| {
+		// Solve the DustLost first
+		let _ = pallet_balances::Pallet::<Runtime>::deposit_creating(
+			&sibling_account(2),
+			1_000_000_000_000,
+		);
+
+		// Sending 10 ParaA token after xcm fee to BOB by XTokens::transfer_multiasset
+		assert_ok!(XTokens::transfer_multiasset(
+			Origin::signed(AccountId::from(ALICE)),
+			Box::new(
+				MultiAsset {
+					id: Concrete(para_native_token_multilocation(1)),
+					fun: Fungible((UnitWeightCost::get() * 4 + 10).into())
```

Comment:
Any specific reason to use this number `UnitWeightCost::get() * 4 + 10`? Can't we use some constant like 0,5 * UNIT?

Reply:
changed